client/container_wait_test.go

```diff
@@ -127,3 +128,36 @@ func ExampleClient_ContainerWait_withTimeout() {
 		log.Fatal(err)
 	}
 }
+
+type TestClient_ContainerWait_cancelReader struct{}
+
+func (TestClient_ContainerWait_cancelReader) Read(p []byte) (n int, err error) {
+	return 0, context.Canceled
+}
+
+func TestClient_ContainerWait_canceled(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	client := &Client{
+		version: "1.30",
+		client: newMockClient(func(req *http.Request) (*http.Response, error) {
+			return &http.Response{
+				StatusCode: http.StatusOK,
+				Body:       io.NopCloser(TestClient_ContainerWait_cancelReader{}),
+			}, nil
+		}),
+	}
+	resultC, errC := client.ContainerWait(ctx, "container_id", "")
+	select {
+	case err := <-errC:
+		if !errors.Is(err, context.Canceled) {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+		return
+	case result := <-resultC:
+		t.Fatalf("Unexpected result: %v", result)
+		return
+	}
+	t.Fatal("Unexpected - we should not reach this line")
```

Comment:
My IDE flags this as unreachable code, which is of course expected here; wondering if any linter will complain (otherwise we could consider changing this to just a comment)

Reply:
No problem.